docs/hadoop.adoc

```diff
@@ -1,20 +1,109 @@
 
 [NOTE]
-The examples in this chapter are based on running Spark in local mode. Additional configuration 
-is required when using Spark in standalone mode or when running Spark on YARN or Mesos.
+The examples in this chapter are based on running Spark in local mode or standalone cluster mode. Additional configuration
+is required when using Spark on YARN or Mesos.
+
+=== Configuring Hadoop for Running OLAP
+For running OLAP queries from the Gremlin Console, a few prerequisites need to be fulfilled. You will need to add the Hadoop configuration directory into the `CLASSPATH`, and the configuration directory needs to point to a live Hadoop cluster.
+
+Hadoop provides a distributed access-controlled file system. The Hadoop file system is used by Spark workers running on different machines to have a common source for file based operations. The intermediate computations of various OLAP queries may be persisted on the Hadoop file system.
+
+For configuring a single node Hadoop cluster, please refer to official https://hadoop.apache.org/docs/r$MAVEN{hadoop2.version}/hadoop-project-dist/hadoop-common/SingleCluster.html[Apache Hadoop Docs]
+
+Once you have a Hadoop cluster up and running, we will need to specify the Hadoop configuration files in the `CLASSPATH`. The below document expects that you have those configuration files located under `/etc/hadoop/conf`.
+
+Once verified, follow the below steps to add the Hadoop configuration to the `CLASSPATH` and start the Gremlin Console, which will play the role of the Spark driver program.
+
+[source, shell]
+----
+export HADOOP_CONF_DIR=/etc/hadoop/conf
+export CLASSPATH=$HADOOP_CONF_DIR
+bin/gremlin.sh
+----
+
+Once the path to Hadoop configuration has been added to the `CLASSPATH`, we can verify whether the Gremlin Console can access the Hadoop cluster by following these quick steps:
+
+[source, gremlin]
+----
+gremlin> hdfs
+==>storage[org.apache.hadoop.fs.LocalFileSystem@65bb9029] // BAD
+
+gremlin> hdfs
+==>storage[DFS[DFSClient[clientName=DFSClient_NONMAPREDUCE_1229457199_1, ugi=user (auth:SIMPLE)]]] // GOOD
+----
+
+
+=== OLAP Traversals
+
+JanusGraph-Hadoop works with TinkerPop's hadoop-gremlin package for general-purpose OLAP to traverse over the graph, and parallelize queries by leveraging Apache Spark.
+
+==== OLAP Traversals with Spark Local
+
+The backend demonstrated here is Cassandra for the OLAP example below. Additional configuration will be needed that is specific to that storage backend. The configuration is specified by the `gremlin.hadoop.graphReader` property which specifies the class to read data from the storage backend.
+
+JanusGraph currently supports following graphReader classes:
+
+* `Cassandra3InputFormat` for use with Cassandra 3
+* `CassandraInputFormat` for use with Cassandra 2
+* `HBaseInputFormat` for use with HBase
+
```

Comment:
Can we add 'HBaseSnapshotInputFormat for use with HBase'?  HBaseSnapshotInputFormat is available in 0.3, but not in 0.2.x